play-akka-http-ws-standalone/src/main/java/play/libs/ws/akkahttp/StandaloneAkkaHttpWSResponse.java

```diff
@@ -0,0 +1,211 @@
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import play.libs.ws.BodyReadable;
+import play.libs.ws.StandaloneWSResponse;
+import play.libs.ws.WSCookie;
+import play.libs.ws.WSCookieBuilder;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public final class StandaloneAkkaHttpWSResponse implements StandaloneWSResponse {
+
+  // FIXME make configurable
+  final Duration UNMARSHAL_TIMEOUT = Duration.ofSeconds(1);
+
+  private final HttpResponse response;
+  private final Materializer mat;
+
+  private HttpResponse strictResponse;
+
+  StandaloneAkkaHttpWSResponse(HttpResponse response, Materializer mat) {
+    this.response = response;
+    this.mat = mat;
+  }
+
+  /**
+   * @return all the headers from the response.
+   */
+  @Override
+  public Map<String, List<String>> getHeaders() {
+    final Map<String, List<String>> headers = new HashMap<>();
+    for (final HttpHeader header: response.getHeaders()) {
+      if (headers.containsKey(header.name())) {
+        headers.get(header.name()).add(header.value());
+      }
+      else {
+        headers.put(header.name(), new ArrayList<>(Collections.singletonList(header.value())));
+      }
+    }
+    return headers;
+  }
+
+  /**
+   * @return the underlying implementation response object, if any.
+   */
+  @Override
+  public Object getUnderlying() {
+    return response;
+  }
+
+  /**
+   * @return the HTTP status code from the response.
+   */
+  @Override
+  public int getStatus() {
+    return response.status().intValue();
+  }
+
+  /**
+   * @return the text associated with the status code.
+   */
+  @Override
+  public String getStatusText() {
+    return response.status().reason();
+  }
+
+  /**
+   * @return all the cookies from the response.
+   */
+  @Override
+  public List<WSCookie> getCookies() {
+    return StreamSupport.stream(response.getHeaders().spliterator(), false)
+      .filter((h) -> h instanceof SetCookie)
+      .map((h) -> {
+        final SetCookie c = (SetCookie)h;
+        return new WSCookieBuilder().setName(c.cookie().name()).setValue(c.cookie().value()).build();
+      })
+      .collect(Collectors.toList());
+  }
+
+  /**
+   * @param name the cookie name
+   * @return a single cookie from the response, if any.
+   */
+  @Override
+  public Optional<WSCookie> getCookie(String name) {
+    return getCookies().stream().filter((c) -> c.getName().equals(name)).findFirst();
+  }
+
+  /**
+   * @return the content type.
+   */
+  @Override
+  public String getContentType() {
+    // FIXME JAVA API no CotentTypes.NoContentType Java Api in Akka Http
+    return response.entity().getContentType()
+      .equals(akka.http.scaladsl.model.ContentTypes.NoContentType()) ? null : response.entity().getContentType().toString();
+  }
+
+  /**
```

Comment:
No need to duplicate Java docs here since `javadoc` will do that already. The same applies to other methods and classes.